plugins/outputs/cloudwatch/cloudwatch_test.go

```diff
@@ -59,6 +59,108 @@ func TestBuildDimensions(t *testing.T) {
 	}
 }
 
+// Test that global dimensions are added correctly
+func TestGlobalDimensions(t *testing.T) {
+	assert := assert.New(t)
+
+	testPoint := testutil.TestMetric(1)
+	globalDimensions := map[string]string{
+		"Environment": "test",
+		"Deployment":  "green",
+	}
+	dimensions := BuildDimensions(testPoint.Tags(), globalDimensions)
+
+	tagKeys := make([]string, len(testPoint.Tags()))
+	i := 0
+	for k := range testPoint.Tags() {
+		tagKeys[i] = k
+		i += 1
+	}
+
+	sort.Strings(tagKeys)
+
+	if len(testPoint.Tags()) >= MaxDimensions {
+		assert.Equal(MaxDimensions, len(dimensions), "Number of dimensions should be less than MaxDimensions")
+	} else {
+		assert.Equal(len(testPoint.Tags())+len(globalDimensions), len(dimensions), "Number of dimensions should be equal to number of tags plus the number of global dimensions")
+	}
```

Comment:
Is there ambiguity in the number of dimensions? I haven't actually looked at this test code closely before. I assumed that the test metric is consistent across test executions. I'm not convinced we need to have this if-else check for asserting. We should just do a single assertion for the size. Which would warrant having another case to account for when we exceed the number of dimensions

Reply:
Again, this is the pattern used in the surrounding tests which is why I've adopted it here. I think it was probably originally set up to support Property Based Testing but it doesn't look like that's actually in use. Looking at the coverage, there are no tests for the case where we're exceeding `MaxDimensions` in the pre-existing code (which definitely doesn't mean we shouldn't add the tests here!).

Reply:
I've added some explicit tests around the max dimensions now and I've removed these weird checks for `MaxDimensions` in these tests.